Set up location driven geotriggers (#63)
Esri/arcgis-maps-sdk-kotlin-samples: set-up-location-driven-geotriggers/README.md

# Set up location driven geotriggers

Create a notification every time a given location data source has entered and/or exited a set of features or graphics.

![Geotriggers](set-up-location-driven-geotriggers.png)

## Use case

Geotriggers can be used to notify users when they have entered or exited a geofence by monitoring a given set of features or graphics. They could be used to display contextual information to museum visitors about nearby exhibits, notify hikers when they have wandered off their desired trail, notify dispatchers when service workers arrive at a scene, or more.

## How to use the sample

Observe a virtual walking tour of the Santa Barbara Botanic Garden. Information about the user's current Garden Section, as well as information about nearby points of interest within 10 meters will display or be removed from the UI when the user enters or exits the buffer of each feature.

## How it works

> 1. Create a `GeotriggerFeed` with an `AbstractLocationDataSource` class (in this case, a `SimulatedLocationDataSource`).

Reviewer: We don't have `AbstractLocationDataSource`
```suggestion
1. Create a `GeotriggerFeed` with a `LocationDataSource` class (in this case, a `SimulatedLocationDataSource`).
```

Author: I think I'm missing to refactor the readme to match the API spec. Will add this to my to-do before the PR.

2. Create a `FeatureFenceParameters` class from a `ServiceFeatureTable`, a buffer distance at which to monitor each feature, an Arcade Expression, and a name for the specific geotrigger.
3. Create a `FenceGeotrigger` with the geotrigger feed, a `FenceRuleType`, and the fence parameters.

> 4. Create a `GeotriggerMonitor` with the fence geotrigger and call `GeotriggerMonitor::start()` to begin listening for events that meet the `FenceRuleType`.

Reviewer: ```suggestion
4. Create a `GeotriggerMonitor` with the fence geotrigger and call `GeotriggerMonitor.start()` to begin listening for events that meet the `FenceRuleType`.
```

> 5. When a `GeotriggerMonitor::geotriggerNotification` emits, capture the `GeotriggerNotificationInfo`.

Reviewer: ```suggestion
5. When a `GeotriggerMonitor.geotriggerNotificationEvent` emits, capture the `GeotriggerNotificationInfo`.
```

> 6. For more information about the feature that triggered the notification, cast the `GeotriggerNotificationInfo` to a `FenceGeotriggerNotificationInfo` and call `FenceGeotriggerNotificationInfo::fenceGeoElement()`.

Reviewer: ```suggestion
6. For more information about the feature that triggered the notification, cast the `GeotriggerNotificationInfo` to a `FenceGeotriggerNotificationInfo` and use its `FenceGeotriggerNotificationInfo.fenceGeoElement`.
```

> 7. Depending on the `FenceGeotriggerNotificationInfo::fenceNotificationType()` display or hide information on the UI from the `GeoElement`'s attributes.

Reviewer: ```suggestion
7. Depending on the `FenceGeotriggerNotificationInfo.fenceNotificationType` display or hide information on the UI from the `GeoElement`'s attributes.
```

## How it works

> 1. Create a `GeotriggerFeed` with an `LocationDataSource` class (in this case, a `SimulatedLocationDataSource`).

Reviewer: ```suggestion
1. Create a `GeotriggerFeed` with a `LocationDataSource` class (in this case, a `SimulatedLocationDataSource`).
```

2. Create a `FeatureFenceParameters` class from a `ServiceFeatureTable`, a buffer distance at which to monitor each feature, an Arcade Expression, and a name for the specific geotrigger.
3. Create a `FenceGeotrigger` with the geotrigger feed, a `FenceRuleType`, and the fence parameters.
4. Create a `GeotriggerMonitor` with the fence geotrigger and call `GeotriggerMonitor.start()` to begin listening for events that meet the `FenceRuleType`.

> 5. When a `GeotriggerMonitor.geotriggerNotification` emits, capture the `GeotriggerNotificationInfo`.

Reviewer: ```suggestion
5. When `GeotriggerMonitor.geotriggerNotification` emits, capture the `GeotriggerNotificationInfo`.
```
